qcodes/instrument/visa.py

```diff
@@ -139,12 +141,22 @@ def device_clear(self) -> None:
         if self.visabackend == 'sim':
             return
 
-        if isinstance(self.visa_handle, pyvisa.resources.SerialInstrument):
-            self.visa_handle.flush(
-                vi_const.BufferOperation.discard_read_buffer_no_io | vi_const.BufferOperation.discard_write_buffer
+        assert isinstance(self.visa_handle, pyvisa.resources.SerialInstrument)
```

Comment:
This assert will not in general be true. This code can be called for all instrument types.